File: solutions/c/rna-transcription/1/rna_transcription.c
#include "rna_transcription.h"
#include <string.h>
#include <stdlib.h>

char *to_rna(const char *dna) {
    size_t n = strlen(dna);
    char *res = (char*)malloc(sizeof(char) * n);
    for (size_t i = 0; i < n; ++i) {
        if (dna[i] == 'G')
            res[i] = 'C';
        else if (dna[i] == 'C')
            res[i] = 'G';
        else if (dna[i] == 'T')
            res[i] = 'A';
        else if (dna[i] == 'A')
            res[i] = 'U';
    }
    return res;
}

File: solutions/go/rna-transcription/1/rna_transcription.go
package strand

func ToRNA(dna string) string {
	res := make([]byte, 0)
    for i := 0; i < len(dna); i++ {
        if dna[i] == 'G' {
            res = append(res, 'C')
        } else if dna[i] == 'C' {
            res = append(res, 'G')
        } else if dna[i] == 'T' {
            res = append(res, 'A')
        } else if dna[i] == 'A' {
            res = append(res, 'U')
        }
    }
    return string(res)
}
